File: backend/tour-design-api/src/main/java/com/tourdesign/platform/model/RestaurantModel.java
package com.tourdesign.platform.model;

import lombok.*;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RestaurantModel {
    private Long id;

    private String name;
    private String cuisineType;
    private String location;
    private Double averagePrice;
    private Integer capacity;
    private String services;

    private List<PackageRestaurantModel> packages;
}

File: frontend/tour-design-web/src/app/models/restaurant.model.ts
// src/app/models/restaurant.model.ts
import { PackageRestaurantModel } from './package-restaurant.model';

export interface RestaurantModel {
  id: number;
  name: string;
  cuisineType?: string;
  location?: string;
  averagePrice?: number;
  capacity?: number;
  services?: string;

  packages?: PackageRestaurantModel[];
}
